### ISSUE TITLE
Solution of Lagrange vs Mayer problems

### ISSUE THREAD

**Issue by ocots** [control-toolbox/CTDirect.jl#82]
https://github.com/control-toolbox/CTDirect.jl/blob/be677bc83fea8afcb31b028dba029bf86f562493/src/solution.jl#L43

In the case of a Lagrange problem, do not return the augmented state. What is it returned?

**Comment by PierreMartinon** [comment]
Good point, at the moment I do not recall a particular reason to return also the additional state for the running cost. When doing a warmstart we do not use these values anyway
https://github.com/control-toolbox/CTDirect.jl/blob/be677bc83fea8afcb31b028dba029bf86f562493/src/init.jl#L58

I suspect it was more convenient for the parsing of the discretized solution. Should we interpolate only the OCP state ?
```
x = ctinterpolate(T, matrix2vec(X[:,1:docp.ocp.control_dimension], 1)) 
```

**Comment by ocots** [comment]
Yes we can do it only when returning the solution. It should be done also for the adjoint and others. I have an example where they have to write $ x[1:n](t) $ and $ p[1:n](t) $ to compute the control $ u(t) = u_s(x[1:n](t), p[1:n](t)) $. It would be better to write $u(t) = u_s(x(t), p(t))$.

**Comment by PierreMartinon** [comment]
Ok, I'll take care of this. I think we'll do a release soon, tell me if you find other things to fix, I'll review the issues list.

**Comment by PierreMartinon** [comment]
Closing for now, reopen if needed.
